### ISSUE TITLE
Lost intellisense / auto-complete

### ISSUE THREAD

**Issue by dstpierre** [Microsoft/vscode-go#167]
Just installed the new version 0.6.21.  Some package simply lost intellisense. Stripe and my own data access package, which were working fine.

I tried uninstalling, closing all instance of VSCode, re-installing, restart etc.

Any idea?
Thanks

**Comment by dstpierre** [comment]
I'm running vscode v. 0.10.5 on Windows 10. I tried to removed the extension folder `rm -rf ~/.vscode/extensions/lukehoban.Go.0.6.21` and the node_modules folder cannot be deleted from there. I uninstall via vscode and re-install. Still no luck.

I will try from my Arch vm to compare. This is brutal at the moment though.

**Comment by dstpierre** [comment]
After some digging it appears that `gocode` is not receiving my GOPATH anymore. After opening an issue on the `gocode` project it appears that `gocode` should be sent the GOPATH value. Could that be the last changes in `goSuggest.ts` at line 61, the addition of the `env` object resetting GOOS and GOARCH that now make the GOPATH to empty? Here's the log for the `gocode -debug -s`

```
2015/12/23 07:26:11 Import path "github.com/stripe/stripe-go" was not resolved
2015/12/23 07:26:11 Gocode's build context is:
2015/12/23 07:26:11  GOROOT: c:\go
2015/12/23 07:26:11  GOPATH:
2015/12/23 07:26:11  GOOS: windows
2015/12/23 07:26:11  GOARCH: amd64
2015/12/23 07:26:11  GBProjectRoot: ""
2015/12/23 07:26:11  lib-path: ""
```

This issue is so annoying I think I will try to build a local version of the extension and try to remove or add the GOPATH in the `env` object, but maybe someone can confirm that it would be that in fact.

**Comment by dstpierre** [comment]
I tested locally and I can confirmed that when adding this line in goSetting.ts at line 61 it's working:

```javascript
var p = cp.execFile(gocode, ["-f=json", "autocomplete", filename, "c" + offset], {
  ...
  GOOS: "",
  GOARCH: "",
  GOPATH: process.env.GOPATH
}
				}
```

Are the GOPATH is sent by default elsewhere, am I the only one having the behaviour?

**Comment by smithfox** [comment]
I met the same issue

**Comment by smithfox** [comment]
I guess the issue only can be found on windows

**Comment by lukehoban** [comment]
@dstpierre Thanks so much for investigating this, and sorry about the issue.  This should be fixed now with 28050d9268f1657f39ac4b0c285e0936e33a7a0c, but I wasn't able to reproduce the problem myself, even on Windows, to verify for sure that the fix you mention really does address this.  I've pushed `0.6.22` with this fix, so you could try that out and re-open if you are still seeing a problem here?
